File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

/**
 * 购物车接口
 */
public interface CartService {
    /**
     * 添加商品到购物车列表
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);

    /**
     * 根据用户名从Redis中查询购物车列表
     * @param username
     * @return
     */
    List<Cart> findCartListFromRedis(String username);

    /**
     * 保存购物车到Redis
     * @param username
     * @param cartList
     */
    void saveCartListToRedis(String username,List<Cart> cartList);

    /**
     * 合并购物车
     * @param cookieCartList
     * @param redisCartList
     * @return
     */
    List<Cart> mergeCartList(List<Cart> cookieCartList,List<Cart> redisCartList);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
//购物车服务层
app.service('cartService',function($http){
    //购物车列表
    this.findCartList=function(){
        return $http.get('cart/findCartList.do');
    }

    //添加商品到购物车
    this.addGoodsToCartList=function(itemId,num){
        return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
    }

    //商品属性合计和商品总计
    this.num = function (cartList) {
        var totalValue = {totalNum:0,totalPrice:0.00};
        for (var i=0;i<cartList.length;i++){
            var cart = cartList[i];
            for (var j=0;j<cart.orderItemsList.length;j++){
                var orderItem = cart.orderItemsList[j];
                totalValue.totalNum += orderItem.num;
                totalValue.totalPrice += orderItem.totalFee;
            }
        }
        return totalValue;
    }
});